File: Data Structures & Algorithms/daily-temperatures/submission-3.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temp) {
        vector<int>ans(temp.size());
        for(int i=temp.size()-2;i>=0;--i){
            int j = i+1;
            while(j<temp.size() and temp[i]>=temp[j]){
                if(ans[j]==0){
                    j=INT_MAX;break;
                }else{
                    j+=ans[j];
                }
            }
            if(j<temp.size()){
                ans[i] = j-i;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-3.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = {}
        for s in strs:
            frq = [0]*26
            for char in s:
                frq[ord(char)-ord('a')] += 1
            if tuple(frq) in res:
                res[tuple(frq)].append(s)
            else:
                res[tuple(frq)]=[s]
        ans = []
        for k,v in res.items():
            ans.append(v)
        return ans